### DISCUSSION TITLE
Compability with lspinstall

### DISCUSSION THREAD

**Discussion by callebstrom** [mfussenegger/nvim-jdtls#121] [Q&A] [answered]
Hi!

I've been using your plugin with success for quite some time. However, as I migrated to [`nvim-lspinstall`](https://github.com/kabouzeid/nvim-lspinstall), in order to reduce the amount of custom configuration, I started running into trouble.

I essentially install and start the java jdtls server via `nvim-lspinstall` default configuration. This means I do not start jdtls via `nvim-jdtls` anymore, but still want to utilise `nvim-jdtls` for the `dap` commands. 

From what I've gather looking through the source this seems like it should be compatible as `nvim-jdtls` resolves the attached jdtls client from the current buffer. However, I end up getting [this](https://github.com/mfussenegger/nvim-jdtls/blob/e04105f551a982663b8d7707a064b733ab71db9b/lua/jdtls/dap.lua#L162) error message which seems to indicate that I am after all not trying to execute the `vscode.java.test.findTestTypesAndMethods` command against the jdtls server initialised via `nvim-lspinstall`.

I have both `com.microsoft.java.test.plugin-0.31.2.jar` and `com.microsoft.java.debug.plugin-0.32.0.jar` in my `bundles` configuration and these are recognised and loaded according to the `lsp.log`.

With all that said, do you see any obvious reasons this would not work given the two JDT plugins above?

Thanks!

**Comment by callebstrom** [comment, marked as answer]
Turns out I had misconfigured `dap`. `nvim-jdtls` works perfectly with `dap` + `nvim-lspinstall`. Ignore me!

**Reply by mfussenegger** [reply]
Glad you got it working. Just keep in mind that I would recommend to still use the `jdtls` setup function. (You should be able to re-use the `cmd` definition of `nvim-lspinstall`. 

The `jdtls` setup function sets some extra options and I'm generally operating under the assumption that these options are set.
